{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "adb73493",
   "metadata": {},
   "outputs": [],
   "source": [
    "import xlrd\n",
    "import pandas as pd\n",
    "import pickle5 as pickle #pickle5 or pickle?\n",
    "from libsbml import *\n",
    "from pprint import pprint\n",
    "from cobra.flux_analysis import gapfill\n",
    "from copy import *\n",
    "from cobra import *\n",
    "from cobra.flux_analysis import *\n",
    "from time import time\n",
    "from riptide import *\n",
    "\n",
    "import cobra\n",
    "import csv\n",
    "import numpy as np\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "f242ba27",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/lib/python3.9/site-packages/cobra/io/dict.py:89: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.\n",
      "Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n",
      "  if isinstance(value, np.float):\n",
      "/opt/anaconda3/lib/python3.9/site-packages/cobra/io/dict.py:91: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.\n",
      "Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations\n",
      "  if isinstance(value, np.bool):\n"
     ]
    }
   ],
   "source": [
    "model =cobra.io.read_sbml_model(\"/Users/varunpasupuleti/Documents/CSBL/iPau21.xml\")   #If .xml file\n",
    "cobra.io.save_json_model(model, \"iPau21.json\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "f101c1e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "LB_open = {'EX_cpd00001_e','EX_cpd00009_e', 'EX_cpd00011_e','EX_cpd00021_e','EX_cpd00034_e','EX_cpd00048_e' ,'EX_cpd00058_e',\n",
    "           'EX_cpd00205_e' ,'EX_cpd00254_e','EX_cpd00971_e','EX_cpd01012_e','EX_cpd00067_e','EX_cpd00528_e'}\n",
    "    \n",
    "LB_limited = {'EX_cpd00023_e'     ,'EX_cpd00027_e'    ,'EX_cpd00033_e'   \n",
    "              ,'EX_cpd00035_e'    ,'EX_cpd00039_e'    ,'EX_cpd00041_e'   \n",
    "              ,'EX_cpd00051_e'    ,'EX_cpd00054_e'    ,'EX_cpd00060_e'   \n",
    "              ,'EX_cpd00065_e'    ,'EX_cpd00066_e'    ,'EX_cpd00069_e'   \n",
    "              ,'EX_cpd00084_e'    ,'EX_cpd00107_e'    ,'EX_cpd00119_e'    \n",
    "              ,'EX_cpd00129_e'    ,'EX_cpd00156_e'    ,'EX_cpd00161_e'  \n",
    "              ,'EX_cpd00305_e'    ,'EX_cpd00322_e'    ,'EX_cpd00092_e'   \n",
    "              ,'EX_cpd00307_e'    ,'EX_cpd03091_e'   }\n",
    "    \n",
    "    \n",
    "LB_ox = {'EX_cpd00007_e'}\n",
    "\n",
    "LB_0_01 = {'EX_cbl1_e'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "52f368e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "for x in model.boundary: \n",
    "    x.lower_bound = 0.0\n",
    "    x.upper_bound = 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "16dd1878",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EX_cpd00034_e (-1000.0, 1000.0)\n",
      "EX_cpd00528_e (-1000.0, 1000.0)\n",
      "EX_cpd00021_e (-1000.0, 1000.0)\n",
      "EX_cpd00011_e (-1000.0, 1000.0)\n",
      "EX_cpd00067_e (-1000.0, 1000.0)\n",
      "EX_cpd00058_e (-1000.0, 1000.0)\n",
      "EX_cpd00009_e (-1000.0, 1000.0)\n",
      "EX_cpd00205_e (-1000.0, 1000.0)\n",
      "EX_cpd00048_e (-1000.0, 1000.0)\n",
      "EX_cpd00001_e (-1000.0, 1000.0)\n",
      "EX_cpd00971_e (-1000.0, 1000.0)\n",
      "EX_cpd01012_e (-1000.0, 1000.0)\n",
      "EX_cpd00254_e (-1000.0, 1000.0)\n",
      "\n",
      "number changed = \n",
      "13\n",
      "number not changed = \n",
      "0\n"
     ]
    }
   ],
   "source": [
    "number_changed = 0\n",
    "number_not_included = 0\n",
    "not_included = \"\"\n",
    "\n",
    "for reaction in LB_open: \n",
    "    previous_number_changed = number_changed\n",
    "    for x in model.boundary: \n",
    "        if x.id == reaction: \n",
    "            x.lower_bound = -1000.0\n",
    "            x.upper_bound = 1000.0\n",
    "            print(x.id, x.bounds)\n",
    "            number_changed = number_changed + 1\n",
    "    if number_changed == previous_number_changed:\n",
    "        not_included = not_included + reaction + \", \"\n",
    "        number_not_included = number_not_included + 1\n",
    "\n",
    "print(\"\")\n",
    "print(\"number changed = \")\n",
    "print(number_changed)\n",
    "print(\"number not changed = \")\n",
    "print(number_not_included)\n",
    "if number_not_included != 0:\n",
    "    print(\"reactions not included = \")\n",
    "    print(not_included)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "e3380abf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EX_cpd00156_e (-10.0, 10.0)\n",
      "EX_cpd00033_e (-10.0, 10.0)\n",
      "EX_cpd00065_e (-10.0, 10.0)\n",
      "EX_cpd00054_e (-10.0, 10.0)\n",
      "EX_cpd00060_e (-10.0, 10.0)\n",
      "EX_cpd00161_e (-10.0, 10.0)\n",
      "EX_cpd00092_e (-10.0, 10.0)\n",
      "EX_cpd00035_e (-10.0, 10.0)\n",
      "EX_cpd00084_e (-10.0, 10.0)\n",
      "EX_cpd00051_e (-10.0, 10.0)\n",
      "EX_cpd00307_e (-10.0, 10.0)\n",
      "EX_cpd00069_e (-10.0, 10.0)\n",
      "EX_cpd00322_e (-10.0, 10.0)\n",
      "EX_cpd00039_e (-10.0, 10.0)\n",
      "EX_cpd00027_e (-10.0, 10.0)\n",
      "EX_cpd00066_e (-10.0, 10.0)\n",
      "EX_cpd00129_e (-10.0, 10.0)\n",
      "EX_cpd00023_e (-10.0, 10.0)\n",
      "EX_cpd03091_e (-10.0, 10.0)\n",
      "EX_cpd00305_e (-10.0, 10.0)\n",
      "EX_cpd00041_e (-10.0, 10.0)\n",
      "EX_cpd00119_e (-10.0, 10.0)\n",
      "EX_cpd00107_e (-10.0, 10.0)\n",
      "\n",
      "number changed = \n",
      "23\n",
      "number not changed = \n",
      "0\n"
     ]
    }
   ],
   "source": [
    "number_changed = 0\n",
    "number_not_included = 0\n",
    "not_included = \"\"\n",
    "\n",
    "for reaction in LB_limited: \n",
    "    previous_number_changed = number_changed\n",
    "    for x in model.boundary: \n",
    "        if x.id == reaction: \n",
    "            x.lower_bound = -10.0\n",
    "            x.upper_bound = 10.0\n",
    "            print(x.id, x.bounds)\n",
    "            number_changed = number_changed + 1\n",
    "    if number_changed == previous_number_changed:\n",
    "        not_included = not_included + reaction + \", \"\n",
    "        number_not_included = number_not_included + 1\n",
    "\n",
    "print(\"\")\n",
    "print(\"number changed = \")\n",
    "print(number_changed)\n",
    "print(\"number not changed = \")\n",
    "print(number_not_included)\n",
    "if number_not_included != 0:\n",
    "    print(\"reactions not included = \")\n",
    "    print(not_included)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "b4dcff2f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EX_cpd00007_e (-20, 1000.0)\n",
      "\n",
      "number changed = \n",
      "1\n",
      "number not changed = \n",
      "0\n"
     ]
    }
   ],
   "source": [
    "number_changed = 0\n",
    "number_not_included = 0\n",
    "not_included = \"\"\n",
    "\n",
    "for reaction in LB_ox: \n",
    "    previous_number_changed = number_changed\n",
    "    for x in model.boundary: \n",
    "        if x.id == reaction: \n",
    "            x.lower_bound = -20\n",
    "            x.upper_bound = 1000.0\n",
    "            print(x.id, x.bounds)\n",
    "            number_changed = number_changed + 1\n",
    "    if number_changed == previous_number_changed:\n",
    "        not_included = not_included + reaction + \", \"\n",
    "        number_not_included = number_not_included + 1\n",
    "\n",
    "print(\"\")\n",
    "print(\"number changed = \")\n",
    "print(number_changed)\n",
    "print(\"number not changed = \")\n",
    "print(number_not_included)\n",
    "if number_not_included != 0:\n",
    "    print(\"reactions not included = \")\n",
    "    print(not_included)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "89d1acb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir(\"/Users/varunpasupuleti/Documents/CSBL/TSVFiles\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "4812b28a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#skip SRR1056147 and SRR1056133 and SRR1056148\n",
    "sequences = ['SRR']*139 #skipping SRR#+92, so that is why it is 138 not 139\n",
    "for i in range(139):\n",
    "    num=1056149+i\n",
    "    num=str(num)\n",
    "    sequences[i]=sequences[i]+num\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "0006be6c",
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'SRR1056181.tsv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Input \u001b[0;32mIn [34]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28mlocals\u001b[39m()[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdictionary\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(seq)] \u001b[38;5;241m=\u001b[39m {}\n\u001b[1;32m      3\u001b[0m counter \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0\u001b[39m\n\u001b[0;32m----> 5\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mstr\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mseq\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m.tsv\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mr\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m \u001b[38;5;28;01mas\u001b[39;00m reads:\n\u001b[1;32m      6\u001b[0m         \u001b[38;5;28;01mfor\u001b[39;00m line \u001b[38;5;129;01min\u001b[39;00m reads:\n\u001b[1;32m      7\u001b[0m             line \u001b[38;5;241m=\u001b[39m line\u001b[38;5;241m.\u001b[39msplit()\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'SRR1056181.tsv'"
     ]
    }
   ],
   "source": [
    "for seq in sequences:\n",
    "    locals()[\"dictionary\" + str(seq)] = {}\n",
    "    counter = 0\n",
    "    \n",
    "    with open(str(seq) + '.tsv', 'r') as reads:\n",
    "            for line in reads:\n",
    "                line = line.split()\n",
    "                gene_len = float(line[1])\n",
    "                hits = float(line[2])\n",
    "                if hits != 0:\n",
    "                    normalized_numb = (hits * 51) / gene_len\n",
    "                    seq_name = line[0]\n",
    "                    seq_name = 'PA14_' + seq_name[-5:]\n",
    "                    locals()[\"dictionary\"+str(seq)].update( {seq_name : normalized_numb} )\n",
    "                    counter = counter + 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "155dbf60",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 369 from 1588 (76.76% change)\n",
      "Metabolites pruned to 379 from 1325 (71.4% change)\n",
      "Flux through the objective DECREASED to ~2.15 from ~4.26 (49.53% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.189, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 7 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 370 from 1588 (76.7% change)\n",
      "Metabolites pruned to 383 from 1325 (71.09% change)\n",
      "Flux through the objective DECREASED to ~2.23 from ~4.26 (47.65% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.173, p=0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 7 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 364 from 1588 (77.08% change)\n",
      "Metabolites pruned to 376 from 1325 (71.62% change)\n",
      "Flux through the objective DECREASED to ~2.14 from ~4.26 (49.77% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.211, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 8 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 365 from 1588 (77.02% change)\n",
      "Metabolites pruned to 378 from 1325 (71.47% change)\n",
      "Flux through the objective DECREASED to ~2.31 from ~4.26 (45.77% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.165, p=0.002 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 9 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 363 from 1588 (77.14% change)\n",
      "Metabolites pruned to 375 from 1325 (71.7% change)\n",
      "Flux through the objective DECREASED to ~2.35 from ~4.26 (44.84% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.191, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 9 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 365 from 1588 (77.02% change)\n",
      "Metabolites pruned to 376 from 1325 (71.62% change)\n",
      "Flux through the objective DECREASED to ~2.15 from ~4.26 (49.53% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.226, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 8 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 365 from 1588 (77.02% change)\n",
      "Metabolites pruned to 376 from 1325 (71.62% change)\n",
      "Flux through the objective DECREASED to ~2.15 from ~4.26 (49.53% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.204, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 8 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 373 from 1588 (76.51% change)\n",
      "Metabolites pruned to 384 from 1325 (71.02% change)\n",
      "Flux through the objective DECREASED to ~2.16 from ~4.26 (49.3% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.257, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 11 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 368 from 1588 (76.83% change)\n",
      "Metabolites pruned to 381 from 1325 (71.25% change)\n",
      "Flux through the objective DECREASED to ~2.3 from ~4.26 (46.01% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.167, p=0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 7 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 372 from 1588 (76.57% change)\n",
      "Metabolites pruned to 385 from 1325 (70.94% change)\n",
      "Flux through the objective DECREASED to ~2.18 from ~4.26 (48.83% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.154, p=0.003 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 7 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 372 from 1588 (76.57% change)\n",
      "Metabolites pruned to 385 from 1325 (70.94% change)\n",
      "Flux through the objective DECREASED to ~2.15 from ~4.26 (49.53% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.132, p=0.011 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 9 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 367 from 1588 (76.89% change)\n",
      "Metabolites pruned to 378 from 1325 (71.47% change)\n",
      "Flux through the objective DECREASED to ~2.16 from ~4.26 (49.3% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.248, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 13 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 373 from 1588 (76.51% change)\n",
      "Metabolites pruned to 383 from 1325 (71.09% change)\n",
      "Flux through the objective DECREASED to ~2.17 from ~4.26 (49.06% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.18, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 8 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 364 from 1588 (77.08% change)\n",
      "Metabolites pruned to 376 from 1325 (71.62% change)\n",
      "Flux through the objective DECREASED to ~2.41 from ~4.26 (43.43% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.22, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 10 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 373 from 1588 (76.51% change)\n",
      "Metabolites pruned to 386 from 1325 (70.87% change)\n",
      "Flux through the objective DECREASED to ~2.26 from ~4.26 (46.95% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.135, p=0.009 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 12 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 373 from 1588 (76.51% change)\n",
      "Metabolites pruned to 383 from 1325 (71.09% change)\n",
      "Flux through the objective DECREASED to ~2.14 from ~4.26 (49.77% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.262, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 9 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 370 from 1588 (76.7% change)\n",
      "Metabolites pruned to 382 from 1325 (71.17% change)\n",
      "Flux through the objective DECREASED to ~2.13 from ~4.26 (50.0% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.221, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 9 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 365 from 1588 (77.02% change)\n",
      "Metabolites pruned to 376 from 1325 (71.62% change)\n",
      "Flux through the objective DECREASED to ~2.13 from ~4.26 (50.0% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.229, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 8 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 365 from 1588 (77.02% change)\n",
      "Metabolites pruned to 375 from 1325 (71.7% change)\n",
      "Flux through the objective DECREASED to ~2.13 from ~4.26 (50.0% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.194, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 45 seconds \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 369 from 1588 (76.76% change)\n",
      "Metabolites pruned to 380 from 1325 (71.32% change)\n",
      "Flux through the objective DECREASED to ~2.14 from ~4.26 (49.77% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.204, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 9 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 379 from 1588 (76.13% change)\n",
      "Metabolites pruned to 391 from 1325 (70.49% change)\n",
      "Flux through the objective DECREASED to ~2.2 from ~4.26 (48.36% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.272, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 8 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 374 from 1588 (76.45% change)\n",
      "Metabolites pruned to 385 from 1325 (70.94% change)\n",
      "Flux through the objective DECREASED to ~2.22 from ~4.26 (47.89% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.228, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 6 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 386 from 1588 (75.69% change)\n",
      "Metabolites pruned to 398 from 1325 (69.96% change)\n",
      "Flux through the objective DECREASED to ~3.04 from ~4.26 (28.64% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.182, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 9 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 376 from 1588 (76.32% change)\n",
      "Metabolites pruned to 389 from 1325 (70.64% change)\n",
      "Flux through the objective DECREASED to ~2.14 from ~4.26 (49.77% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.241, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 8 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 367 from 1588 (76.89% change)\n",
      "Metabolites pruned to 379 from 1325 (71.4% change)\n",
      "Flux through the objective DECREASED to ~2.16 from ~4.26 (49.3% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.262, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 7 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 364 from 1588 (77.08% change)\n",
      "Metabolites pruned to 374 from 1325 (71.77% change)\n",
      "Flux through the objective DECREASED to ~2.21 from ~4.26 (48.12% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.19, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 7 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 373 from 1588 (76.51% change)\n",
      "Metabolites pruned to 385 from 1325 (70.94% change)\n",
      "Flux through the objective DECREASED to ~2.14 from ~4.26 (49.77% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.232, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 8 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 368 from 1588 (76.83% change)\n",
      "Metabolites pruned to 379 from 1325 (71.4% change)\n",
      "Flux through the objective DECREASED to ~2.19 from ~4.26 (48.59% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.18, p=0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 7 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 366 from 1588 (76.95% change)\n",
      "Metabolites pruned to 376 from 1325 (71.62% change)\n",
      "Flux through the objective DECREASED to ~2.15 from ~4.26 (49.53% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.182, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 9 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 362 from 1588 (77.2% change)\n",
      "Metabolites pruned to 373 from 1325 (71.85% change)\n",
      "Flux through the objective DECREASED to ~2.15 from ~4.26 (49.53% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.134, p=0.011 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 7 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 366 from 1588 (76.95% change)\n",
      "Metabolites pruned to 377 from 1325 (71.55% change)\n",
      "Flux through the objective DECREASED to ~2.13 from ~4.26 (50.0% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.184, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 10 seconds \n",
      "\n",
      "\n",
      "Initializing model and integrating transcriptomic data...\n",
      "Pruning zero flux subnetworks...\n",
      "Analyzing context-specific flux distributions...\n",
      "\n",
      "Reactions pruned to 366 from 1588 (76.95% change)\n",
      "Metabolites pruned to 377 from 1325 (71.55% change)\n",
      "Flux through the objective DECREASED to ~2.17 from ~4.26 (49.06% change)\n",
      "Context-specific metabolism correlates with transcriptome (r=0.193, p<0.001 *)\n",
      "\n",
      "RIPTiDe completed in, 1 minute and 7 seconds \n",
      "\n"
     ]
    },
    {
     "ename": "IndexError",
     "evalue": "list index out of range",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "Input \u001b[0;32mIn [35]\u001b[0m, in \u001b[0;36m<cell line: 5>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m#troubleshoot errors in this cell, should get riptide file outputs\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;66;03m#get file outputs that are riptide results with 6 files inside each\u001b[39;00m\n\u001b[1;32m      3\u001b[0m \u001b[38;5;66;03m#look at riptide github and figure out what outputs mean, flux sample, flux variability\u001b[39;00m\n\u001b[1;32m      5\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m seq \u001b[38;5;129;01min\u001b[39;00m sequences:\n\u001b[0;32m----> 6\u001b[0m     \u001b[38;5;28mlocals\u001b[39m()[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mpa14_riptide_\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m+\u001b[39m\u001b[38;5;28mstr\u001b[39m(seq)] \u001b[38;5;241m=\u001b[39m \u001b[43mriptide\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcontextualize\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmodel\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmodel\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtranscriptome\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mlocals\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mdictionary\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[38;5;28;43mstr\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mseq\u001b[49m\u001b[43m)\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfraction\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m0.5\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      7\u001b[0m     \u001b[38;5;28mlocals\u001b[39m()[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mpa14_model_\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m+\u001b[39m\u001b[38;5;28mstr\u001b[39m(seq)] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlocals\u001b[39m()[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mpa14_riptide_\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m+\u001b[39m\u001b[38;5;28mstr\u001b[39m(seq)]\u001b[38;5;241m.\u001b[39mmodel\n\u001b[1;32m      8\u001b[0m     \u001b[38;5;28mdir\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124mr\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/Users/varunpasupuleti/Documents/CSBL/TSVFiles\u001b[39m\u001b[38;5;124m\"\u001b[39m\n",
      "File \u001b[0;32m/opt/anaconda3/lib/python3.9/site-packages/riptide/riptide.py:597\u001b[0m, in \u001b[0;36mcontextualize\u001b[0;34m(model, transcriptome, samples, silent, exch_weight, fraction, minimum, conservative, objective, additive, important, direct, set_bounds, tasks, exclude, gpr, threshold, open_exchanges, skip_fva)\u001b[0m\n\u001b[1;32m    594\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    595\u001b[0m         transcriptome[x] \u001b[38;5;241m=\u001b[39m [transcriptome[x]]\n\u001b[0;32m--> 597\u001b[0m transcriptome[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mreplicates\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mint\u001b[39m(\u001b[38;5;28mlen\u001b[39m(transcriptome[\u001b[38;5;28;43mlist\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mtranscriptome\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkeys\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m]\u001b[49m]))\n\u001b[1;32m    598\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m gene \u001b[38;5;129;01min\u001b[39;00m transcriptome\u001b[38;5;241m.\u001b[39mkeys():\n\u001b[1;32m    599\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m gene \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mreplicates\u001b[39m\u001b[38;5;124m'\u001b[39m:\n",
      "\u001b[0;31mIndexError\u001b[0m: list index out of range"
     ]
    }
   ],
   "source": [
    "\n",
    "#troubleshoot errors in this cell, should get riptide file outputs\n",
    "#get file outputs that are riptide results with 6 files inside each\n",
    "#look at riptide github and figure out what outputs mean, flux sample, flux variability\n",
    "\n",
    "for seq in sequences:\n",
    "    locals()[\"pa14_riptide_\"+str(seq)] = riptide.contextualize(model=model, transcriptome=locals()[\"dictionary\"+str(seq)], fraction=0.5)\n",
    "    locals()[\"pa14_model_\"+str(seq)] = locals()[\"pa14_riptide_\"+str(seq)].model\n",
    "    dir= r\"/Users/varunpasupuleti/Documents/CSBL/TSVFiles\"\n",
    "    outFile= 'riptide_results_0.5_' + str(seq)\n",
    "    current = dir + outFile\n",
    "    riptide.save_output(riptide_obj=locals()[\"pa14_riptide_\"+str(seq)], path=current)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
